WMF Framework/EventLoggingService.swift

```diff
@@ -418,12 +289,16 @@ public class EventLoggingService : NSObject, URLSessionDelegate {
             
             var request = URLRequest(url: url)
             request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
-
-            let task = urlSession.dataTask(with: request, completionHandler: { (_, response, error) in
+            let session = onlyWiFi ? self.session.wifiOnlyURLSession : self.session.defaultURLSession
+            let task = session.dataTask(with: request, completionHandler: { (_, response, error) in
                 guard error == nil,
                     let httpResponse = response as? HTTPURLResponse,
                     httpResponse.statusCode / 100 == 2 else {
-                        completion(EventLoggingError.generic)
+                        if let error = error as NSError?, error.domain == NSURLErrorDomain {
+                            completion(EventLoggingError.network)
```

Comment:
Would it make sense to make a wrapper around this dataTask method inside our `Session` so that we can check for these generic errors in one place? `WikidataFetcher` is also calling this method and defining its own generic errors

Reply:
Down the line, yes, but not within the scope of this PR as this is currently called directly on the `URLSession`, not the `Session`